gamemodes/cinema/gamemode/modules/theater/services/sh_youtube.lua

```diff
@@ -30,22 +30,125 @@ function SERVICE:GetKey(url)
+									document.getElementsByTagName('video')[0].volume=to_volume/100;
+									
+									var target_t = target_time();
+									
+									if (target_t < (document.getElementsByTagName('video')[0].duration - 1.0)) {
+										if (Math.abs(document.getElementsByTagName('video')[0].currentTime - target_t) > 2.0) {
+											document.getElementsByTagName('video')[0].currentTime = target_t;
+										}
+									}
+									
+									document.getElementsByTagName('video')[0].play();
+								}
+							}
+							setInterval(think,100);
+							
+							function th_volume(volume){
+								to_volume = volume;
+							};
+							
+							function th_seek(seconds){
+								start_time = timestamp() - seconds;
+							};
+						}
+					]])
+					panel:RunJavascript("if(document.getElementsByClassName('ytp-watermark').length)document.getElementsByClassName('ytp-watermark')[0].remove()") --bottom right watermark
+					panel:RunJavascript("if(document.getElementsByClassName('ytp-show-cards-title').length)document.getElementsByClassName('ytp-show-cards-title')[0].remove()") --title bar
+					panel:RunJavascript("if(document.getElementsByClassName('ytp-paid-content-overlay').length)document.getElementsByClassName('ytp-paid-content-overlay')[0].remove()") --ad?
+					panel:RunJavascript("if(document.getElementsByClassName('ytp-pause-overlay').length)document.getElementsByClassName('ytp-pause-overlay')[0].remove()") --pause overlay
+					panel:RunJavascript("if(document.getElementsByClassName('videowall-endscreen').length)document.getElementsByClassName('videowall-endscreen')[0].remove()") --endscreen
+					panel:RunJavascript("if(document.getElementsByClassName('ytp-gradient-top').length)document.getElementsByClassName('ytp-gradient-top')[0].remove()") --top gradient
+				end
+			end)
+		else
+			panel:EnsureURL("http://swamp.sv/s/cinema/" .. self:GetClass() .. ".html")
+			local str = string.format("th_video('%s',%s);", string.JavascriptSafe(Video:Key()), Video:Duration() > 0 and "false" or "true")
+			panel:QueueJavascript(str)
+		end
+		
+		local fn = panel.ConsoleMessage
         panel.ConsoleMessage = function(a, str)
-            fn(a, str)
-
-            if str:len() > 2 and str:sub(1, 2) == "T:" and tonumber(str:sub(3)) then
-                YOUTUBE_TRUE_START = SysTime() - tonumber(str:sub(3))
-                YOUTUBE_TRUE_START_PING = SysTime()
-            end
+			fn(a, str)
+			if (str == "PAGE LOADED") then
+				enabled = false
+				loaded = true
+			else
+				if str:len() > 2 and str:sub(1, 2) == "T:" and tonumber(str:sub(3)) then
+					YOUTUBE_TRUE_START = SysTime() - tonumber(str:sub(3))
+					YOUTUBE_TRUE_START_PING = SysTime()
+				end
+			end
         end
     end
+	
+    function SERVICE:SetVolume(vol, panel)
+		if (self:IsMature(panel.vid)) then
+			local nvar = RandomString()
+			local str = "var "..nvar.."=setInterval(function(){if(typeof(th_volume)!='undefined'){th_volume("..vol..");clearInterval("..nvar..");}},10);"
```

Comment:
You're using a random variable name so that if this gets runs multiple times, each run will store its value in a different variable. Try using a scope instead https://www.w3schools.com/js/js_scope.asp